contracts/root/WithdrawManager.sol

```diff
@@ -0,0 +1,484 @@
+  }
+
+  /**
+  * @dev Determines the next exit to be processed.
+  * @param _token Asset type to be exited.
+  * @return A tuple of the position and time when this exit can be processed.
+  */
+  function getNextExit(address _token)
+    public
+    view
+    returns (uint256, uint256)
+  {
+    return PriorityQueue(exitsQueues[_token]).getMin();
+  }
+
+  /**
+  * @dev Processes any exits that have completed the exit period.
+  */
+  function processExits(address _token) public {
+    uint256 exitableAt;
+    uint256 utxoPos;
+
+    // retrieve priority queue
+    PriorityQueue exitQueue = PriorityQueue(exitsQueues[_token]);
+
+    // Iterate while the queue is not empty.
+    while (exitQueue.currentSize() > 0) {
+      (exitableAt, utxoPos) = getNextExit(_token);
+
+      // Check if this exit has finished its challenge period.
+      if (exitableAt > block.timestamp) {
+        return;
+      }
+
+      // get withdraw block
+      PlasmaExit memory currentExit = exits[utxoPos];
+
+      // process if NFT exists
+      // If an exit was successfully challenged, owner would be address(0).
+      address exitOwner = ExitNFT(exitNFTContract).ownerOf(utxoPos);
+      if (exitOwner != address(0)) {
+        if (_token == wethToken) {
+          // transfer ETH to token owner if `rootToken` is `wethToken`
+          WETH(wethToken).withdraw(currentExit.amount, exitOwner);
+        } else {
+          // transfer tokens to current contract
+          ERC20(_token).transfer(exitOwner, currentExit.amount);
+        }
+
+        // broadcast withdraw events
+        emit Withdraw(exitOwner, _token, currentExit.amount);
+
+        // Delete owner but keep amount to prevent another exit from the same UTXO.
+        // delete exits[utxoPos].owner;
+
+        // burn NFT
+        ExitNFT(exitNFTContract).burn(exitOwner, utxoPos);
+      }
+
+      // exit queue
+      exitQueue.delMin();
+    }
+  }
+
+  // withdraw tokens
+  function withdraw(
+    uint256 headerNumber,
+    bytes headerProof,
+
+    uint256 blockNumber,
+    uint256 blockTime,
+    bytes32 txRoot,
+    bytes32 receiptRoot,
+    bytes path,
+
+    bytes txBytes,
+    bytes txProof,
+
+    bytes receiptBytes,
+    bytes receiptProof
+  ) public {
+    address rootToken;
+    uint256 receiptAmount;
+
+    (rootToken, receiptAmount) = _processWithdrawReceipt(
+      receiptBytes,
+      path,
+      receiptProof,
+      receiptRoot,
+      msg.sender
+    );
+
+    // process withdraw tx
+    _processWithdrawTx(
+      txBytes,
+      path,
+      txProof,
+      txRoot,
+
+      rootToken,
+      receiptAmount,
+
+      msg.sender
+    );
+
+    // withdraw
+    _withdraw(
+      headerNumber,
+      headerProof,
+
+      blockNumber,
+      blockTime,
+
+      txRoot,
+      receiptRoot,
+
+      rootToken,
+      receiptAmount,
+
+      path,
+      0,
+
+      msg.sender
+    );
+  }
+
+  // withdraw tokens
+  function withdrawTokens(
```

Comment:
This function was used in withdrawal-bridge, but the w-bridger is deprecated. So this function must be removed for the sake of readability.

Reply:
This function is to withdraw without burn (last resort). When user is done with Matic and Matic chain is no longer functional (extreme case)

We will update `withdrawal-bridge` once this pull request is done.

Reply:
make sense :)